Mobile 3713

dpalou opened the pull request.

crazyserver commented on src/core/components/iframe/core-iframe.html:
```diff
@@ -5,7 +5,11 @@
     </iframe>
 
+    <div *ngIf="displayHelp" class="ion-margin ion-text-center ion-text-wrap core-iframe-help">
+        <span (click)="openIframeHelpModal()" aria-haspopup="dialog" role="button">{{ 'core.iframehelp' | translate }}</span>
```
Role button is not enough to correctly handle a11y. 
Explanation found:
The custom control should implement onclick as well as onKeyDown events. Buttons can be activated through spacebar. Hence, if you are adding the role to a custom control, you need to handle these events yourself. Else, the semantic loses its meaning. A screen-reader user will try to activate the button using spacebar, but cannot.

 I would suggest to change it to a default/native button and style it properly or handle onKeyDown (or Up) to check space and enter keys when focused.

crazyserver replied:
Also, when using a custom element you should add tabindex=0 so is keyboard accessible.

And finally, you will also need to add :focus and :hover css. It could be added using span[role="button"]:focus so it solves some others.

crazyserver commented on src/core/directives/format-text.ts:
```diff
@@ -757,6 +761,33 @@ export class CoreFormatTextDirective implements OnChanges {
+
+        const span = document.createElement('span');
+        span.innerHTML = Translate.instant('core.iframehelp');
+        span.setAttribute('role', 'button');
```
Role button is not enough to correctly handle a11y. 
Explanation found:
The custom control should implement onclick as well as onKeyDown events. Buttons can be activated through spacebar. Hence, if you are adding the role to a custom control, you need to handle these events yourself. Else, the semantic loses its meaning. A screen-reader user will try to activate the button using spacebar, but cannot.

 I would suggest to change it to a default/native button and style it properly or handle onKeyDown (or Up) to check space and enter keys when focused.

crazyserver commented on src/core/lang.json:
```diff
@@ -213,6 +215,7 @@
     "openfullimage": "Click here to display the full size image",
     "openinbrowser": "Open in browser",
     "openmodinbrowser": "Open {{$a}} in browser",
+    "opensettings": "Open settings",
```
Please add new strings to langindex.json

crazyserver reviewed (changes requested):
Code looks good but I found minor a11y changes.

dpalou commented:
Thanks Pau,

I modified both HTML codes and I added the strings to langindex. I also had to modify the rule in theme.base.scss. Can you please take a look again?

crazyserver reviewed (approved):
All ok!
https://tracker.moodle.org/browse/MOBILE-3713
